### ISSUE TITLE
Use checksum or hash of file as identifier

### ISSUE THREAD

**Issue by pichfl** [GianniCarlo/Audiobook-Player#64]
Instead of using `self.fileURL.lastPathComponent` we should consider using a hash generated from the binary file. 

That paves the way for future features and also ensures that recognize a file that was added previously.

**Comment by pichfl** [comment]
I agree with not doing this when the book is loaded into the player. I'd do it when the book is added to the library.

My main reasoning for this is that is that a books binary data should not change over time. We could get a bit more clever and hash something else or only parts of the book, I'd just argue that two audio books could share the same file name but have completely different content (ie. when you have a file named "Part 1.m4b").

And yes, the intention was to create a way to recognize files that are added back to the player after being either removed for storage saving or because they never were on the device, for example after a backup/restore or when we implement syncing between devices.

**Comment by GianniCarlo** [comment]
good point about different files sharing the same name. I think my problem was that on every app launch, all the files were processed all over again, and calculating that hash made it slow for each book to appear. This wouldn't be a problem once we start persisting the Book class in some way
